message_coding/templates/_layouts/form_base.html

```diff
@@ -0,0 +1,41 @@
+{% load widget_tweaks %}
+
+        {# borrowed from http://wisercoder.com/django-forms-bootstrap/ #}
+        <form class="form" action="" method="POST" {% if form.is_multipart %}enctype="multipart/form-data"{% endif %}>
+            {% csrf_token %}
+            {{ form.media }}
+            {% if form.errors %}
+                <div class="alert alert-error">
+                    <ul>
+                        {% for error in form.non_field_errors %}
+                            <li>{{ error }}</li>
+                        {% endfor %}
+                    </ul>
+                </div>
+            {% endif %}
+            {% for field in form.visible_fields %}
+                <div class="form-group {{ field.html_name }} {% if field.errors %}error{% endif %}">
+                    <label class="control-label">{{ field.label }}{% if field.field.required %} *{% endif %}</label>
+                    <div class="controls">
+                        {% render_field field class+="form-control" %}
+                        {% if field.help_text %}<span class="help-inline">{{ field.help_text }}</span>{% endif %}
+                        {% if field.errors %}<div class="alert alert-danger" role="alert">{{ field.errors.as_text }}</div>{% endif %}
+                    </div>
```

Comment:
I am guessing this `controls` div is intended to help with layout. I think the [blog post](http://wisercoder.com/django-forms-bootstrap/) you started from for this must have been for Bootstrap 2.x rather than 3, since `help-inline` doesn't seem to exist anymore. The Bootstrap form examples seem to use `<p class="help-block">` for help text, although a div or span may work just as well since the help-block class overrides margins and sets display=block anyway.